keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,958 @@
+- release note
+
+1.24:
+- enable by default
+- allow gate to disable the feature
+- release note
+
+1.26:
+- remove gate
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: ClusterNetworkPolicy
+  - Components depending on the feature gate: kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Enabling the feature by itself has no effect on the cluster.
+Creating a ClusterNetworkPolicy/DefaultNetworkPolicy does have an effect on
+the cluster, however they must be specifically created, which means the
+administrator is aware of the impact.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Once enabled, the feature can be disabled via feature gate. However, disabling
+the feature may cause created cluster-scoped policy resources to be deleted,
+which may impact the security of the cluster. Administrators must make provision
+to secure their cluster by other means before disabling the feature gate.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+n/a
+
+###### Are there any tests for feature enablement/disablement?
+
+n/a
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### Will enabling / using this feature result in any new API calls?
+
+Enabling this feature by itself will have no impact on the cluster and no
+new API calls will be made.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+Enabling this feature will introduce new API types as described in the [design](#design-details)
+section. The supported number of objects per cluster will depend on the individual
+CNI providers who will be responsible to provide the implementation to realize
+these resources.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+n/a
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+n/a
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+n/a
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+n/a
+
+## Implementation History
+
+- 2021-02-18 - Created initial PR for the KEP
+
+## Drawbacks
+
+Securing traffic for a cluster for administrator's use case can get complex.
+This leads to introduction of a more complex set of APIs which could confuse
+users.
+
+## Alternatives
+
+Following alternative approaches were considered:
```

Comment:
Have you evaluated the option to implement it as a CRD instead of a core type, ala Gateway API? https://github.com/kubernetes-sigs/gateway-api#kubernetes-gateway-api

Reply:
Is not the norm, but in-tree API objects has much more requirements and the development process is much slower. Having people that went through this thought process is at least worth asking

Reply:
I would have suggested that we start with a CRD for sure.

Reply:
sounds good. I am thinking a more generic repo `network-policy-api` under `kubernetes-sigs` to evolve related CRDs, for example, CNPs, the v2 NP CRD etc. Let us kick start a discussion around those lines and we will make changes here accordingly and in the mean time, we can continue reviewing the spec

Reply:
+1 @abhiraut I think that will be the most efficient path forward

Reply:
I think this should land as a built-in because it makes sense to be in the same group as NetworkPolicy.  We can start with a CRD in a different group.